app/gui/src/main/scala/org/bitcoins/gui/ContractGUI.scala

```diff
@@ -0,0 +1,274 @@
+  AddDLCSigsAndBroadcastCliCommand,
+  CreateDLCOffer,
+  SignDLCCliCommand
+}
+import org.bitcoins.core.protocol.dlc.models.DLCStatus
+import org.bitcoins.core.protocol.tlv.{
+  ContractInfoV0TLV,
+  DLCAcceptTLV,
+  DLCOfferTLV,
+  DLCSignTLV,
+  LnMessageFactory,
+  OracleAnnouncementV0TLV
+}
+import org.bitcoins.gui.contract.GlobalContractData
+import org.bitcoins.gui.dlc.DLCPaneModel
+import org.bitcoins.gui.dlc.dialog.{
+  AcceptOfferDialog,
+  BroadcastDLCDialog,
+  CreateDLCOfferDialog,
+  DLCDialogContainer,
+  SignDLCDialog,
+  ViewDLCDialog
+}
+import org.bitcoins.gui.util.GUIUtil
+import scalafx.beans.property.StringProperty
+import scalafx.geometry._
+import scalafx.scene.Parent
+import scalafx.scene.control.{
+  ContextMenu,
+  MenuItem,
+  TableColumn,
+  TableView,
+  TextField
+}
+import scalafx.scene.layout._
+
+import java.io.File
+import java.nio.file.Files
+import java.text.SimpleDateFormat
+import java.util.Date
+import scala.util.{Failure, Success}
+
+class ContractGUI(glassPane: VBox) {
+
+  def fetchStartingData(): Unit = {
+    // Nothing to fetch for state yet...
+  }
+
+  private[gui] lazy val model = new ContractGUIModel()
+
+  private lazy val addEventTF = new TextField {
+    styleClass += "title-textfield"
+    promptText = "New Event Hex"
+    onKeyTyped = _ => {
+      val event = model.addEvent(this.text.value.trim)
+      event match {
+        case Some(tup) =>
+          clearEventTF()
+          eventTableView.sort()
+          // Set focus on new item
+          eventTableView.getSelectionModel().select(tup)
+          // Show view
+          showCreateOfferPane(tup._1, tup._2)
+        case None =>
+      }
+    }
+  }
+
+  private def clearEventTF(): Unit = {
+    addEventTF.clear()
+  }
+
+  lazy val addEventHBox = new HBox {
+    styleClass += "small"
+    children = Seq(addEventTF)
+  }
+
+  private lazy val addContractTF = new TextField {
+    styleClass += "title-textfield"
+    promptText = "Contract Hex"
+    onKeyTyped = _ => {
+      val bool = onContractAdded(text.value.trim, None)
+      if (bool) clearContractTF() // Clear on valid data
+      ()
+    }
+  }
+
+  private def clearContractTF(): Unit = {
+    addContractTF.clear()
+  }
+
+  private lazy val fileChooserButton = GUIUtil.getFileChooserButton(file => {
+    val hex = Files.readAllLines(file.toPath).get(0)
+    val bool = onContractAdded(hex, Some(file))
+    if (bool) addContractTF.clear()
+  })
+
+  lazy val addContractHBox = new HBox {
+    styleClass += "small"
+    children = Seq(fileChooserButton, addContractTF)
+  }
+
+  private lazy val eventIdCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Event Id"
+    prefWidth = 160
+    cellValueFactory = { status =>
+      val eventIdStr = status.value._1.eventTLV.eventId
+      new StringProperty(status, "Event Id", eventIdStr)
+    }
+  }
+
+  private val ISO_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'")
```

Comment:
we use a different format in the `ViewDLCDialog` should stick to one universally. Imo ISO is good for passing around data but not the most human readable

Reply:
I used ISO because they order well versus weekday format.